assets/js/components/Site.vue

```diff
@@ -0,0 +1,50 @@
+<template>
+	<div>
+		<div class="row">
+			<div class="d-none d-md-flex col-12 col-md-4 mt-md-4 align-items-end">
+				<p class="h1">{{ state.title || "Home" }}</p>
+			</div>
+			<div class="col-12 col-md-8 mt-md-4" v-if="multi">
+				<SiteDetails v-bind:state="state"></SiteDetails>
+			</div>
+		</div>
+
+		<div class="row d-none d-md-flex border-bottom"></div>
+
+		<div class="row" v-if="!multi">
+			<div class="d-none d-md-block col-md-4"></div>
+			<div class="col-12 col-md-8">
+				<SiteDetails v-bind:state="state"></SiteDetails>
+			</div>
+		</div>
+
+		<Loadpoint
+			v-for="(loadpoint, id) in state.loadpoints"
+			v-bind:id="id"
+			:key="id"
+			:state="loadpoint"
+			:pv="state.gridConfigured"
+			:multi="multi"
+		>
+		</Loadpoint>
+	</div>
+</template>
+
+<script>
+import SiteDetails from "./SiteDetails";
+import Loadpoint from "./Loadpoint";
+import formatter from "../mixins/formatter";
+
+export default {
+	name: "Site",
+	props: ["state"],
+	components: { SiteDetails, Loadpoint },
+	mixins: [formatter],
+	computed: {
+		multi: function () {
+			// TODO fix compact
+			return this.state.loadpoints.length > 1 /* || app.compact*/;
```

Comment:
@andig Kannst du mir erklären was der Compact-Mode ist? Habe gesehen, dass das scheinbar aus der URL gezogen wird. Habs hier erstmal auskommentiert weils die globale `app` Variable nicht mehr gibt.

Reply:
"compact" biegt die Darstellung auf den kompakteren "mehrere Ladepunkte" Modus um. Genutzt um das UI auf das 800x400 Display der OpenWB ohne scrollen zu bekommen. Das könnten wir auch später wieder einbauen.

Was ich aber nicht verstanden habe: warum das Du die app vom Singleton in eine Komponente verwandelt- ist Singleton nicht für die Root Komponente ok?

Reply:
Die globale `app` Instanzvariable hab ich entfernt weil man die eigentlich nicht braucht. Kommunikation innerhalb der App findet idR. durch Props und Events statt, nicht durch globale Referenzen. Macht das ganze auch einfacher nachvollziehbarer und testbar.

Die Toasts hätte ich auch gerne in die eigentliche App integriert, aber da das nicht ganz so trivial war hab ich's hier im PR erstmal sein gelassen und die Funktionalität über window.toasts wieder bereit gestellt.